app/models/notification.rb

```diff
@@ -0,0 +1,55 @@
+class Notification < ActiveRecord::Base
```

Comment:
The design of your notification class is incredibly awkward:

 - methods are named X_and_Y, and everything sends an email anyway. Shouldn't the email be optional?
 - All the views are are dumped in a common notifications directory. How does this design scale for a large application with multiple components like Coursemology?
 - How do you know your design will work for all the existing notifications that Coursemology v1 would require to make?